Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,288 @@
+  simp_rw [subset_def, mem_preimage] at hf3
+  /- prove that `γ := (extChartAt I x₀).symm ∘ f` is a desired integral curve -/
+  refine' ⟨(extChartAt I x₀).symm ∘ f, _, min ε1 ε2, lt_min hε1 hε2, _⟩
+  · apply Eq.symm
+    rw [Function.comp_apply, hf1, LocalEquiv.left_inv _ (mem_extChartAt_source ..)]
+  intros t ht
+  /- collect useful terms in convenient forms -/
+  rw [←Real.ball_eq_Ioo] at ht
+  have ht1 := mem_of_mem_of_subset ht (Metric.ball_subset_ball (min_le_left ..))
+  have ht2 := mem_of_mem_of_subset ht (Metric.ball_subset_ball (min_le_right ..))
+  have h : HasDerivAt f
+    ((fderivWithin ℝ ((extChartAt I x₀) ∘ (extChartAt I ((extChartAt I x₀).symm (f t))).symm)
+        (range I) (extChartAt I ((extChartAt I x₀).symm (f t)) ((extChartAt I x₀).symm (f t))))
+      (v ((extChartAt I x₀).symm (f t))))
+    t := hf2 t ht1
+  have hf3' := mem_of_mem_of_subset (hf3 t ht2) interior_subset
+  /- express the derivative of the integral curve in the local chart -/
+  rw [HasMFDerivAt]
+  use ContinuousAt.comp
+    (continuousAt_extChartAt_symm'' _ _ hf3') ((hf2 t ht1).continuousAt)
+  apply HasDerivWithinAt.hasFDerivWithinAt
+  rw [modelWithCornersSelf_coe, range_id, hasDerivWithinAt_univ, ext_chart_model_space_apply,
+    writtenInExtChartAt, Function.comp_apply, Function.comp.assoc, extChartAt_model_space_eq_id,
+    LocalEquiv.refl_symm, LocalEquiv.refl_coe, Function.comp.right_id, ←Function.comp.assoc]
+  /- `h` gives the derivative of `f` at `t` as `↑D (v (γ t))`, where `D` is the change of
+    coordinates from the chart at `γ t` to the chart at `x₀`. we wish to use
+    `HasFDerivAt.comp_hasDerivAt` to get the derivative of `γ` at `t` as `v (γ t)`, which requires
+    first expressing `v (γ t)` as `↑D_inv ↑D (v (γ t))`, where `D_inv` is the opposite change of
+    coordinates as `D`. -/
+  have hvsub : v ((extChartAt I x₀).symm (f t)) = (tangentBundleCore I M).coordChange
+    (achart H x₀) (achart H ((extChartAt I x₀).symm (f t))) ((extChartAt I x₀).symm (f t))
+    ((tangentBundleCore I M).coordChange (achart H ((extChartAt I x₀).symm (f t))) (achart H x₀)
+      ((extChartAt I x₀).symm (f t)) (v ((extChartAt I x₀).symm (f t)))) := by
+    rw [(tangentBundleCore I M).coordChange_comp, (tangentBundleCore I M).coordChange_self]
+    · rw [tangentBundleCore_baseSet, coe_achart, ←extChartAt_source I, ←LocalEquiv.symm_target]
+      exact mem_extChartAt_source ..
+    · rw [tangentBundleCore_baseSet, tangentBundleCore_baseSet, coe_achart, coe_achart,
+        ←extChartAt_source I, ←extChartAt_source I, inter_comm, ←inter_assoc, inter_self]
+      constructor
+      · exact mem_extChartAt_source ..
+      · rw [←mem_preimage]
+        apply mem_of_mem_of_subset _ (LocalEquiv.source_subset_preimage_target _)
+        rw [LocalEquiv.symm_source]
+        exact hf3'
+  rw [hvsub]
+  apply HasFDerivAt.comp_hasDerivAt _ _ h
+  /- change of coordinates in the tangent bundle is exactly the derivative of composition of local
```

Comment:
Didn't think about this in depth yet, but: would this be worth extracting into a separate lemma?

Reply:
Yes. I suggest adding a few lemmas in `TangentBundle` to make working with `coordChange` more straightforward. Will be away from mathlib for a week or so as it's Thanksgiving over here.